src/transformers/cache_utils.py

```diff
@@ -618,6 +616,8 @@ def update(
 
         # Update the cache
         if len(self.key_cache) <= layer_idx:
+            raise ValueError("OffloadedCache does not support model usage where layers are skipped. Use DynamicCache.")
```

Comment:
Classes that expand dynamic cache don't support skipping layers, at least for now. 

This PR did not add this limitation, but adds the exception :)